shared/team-building/email-input/index.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react'
+import {debounce} from 'lodash-es'
+import * as Container from '../../util/container'
+import * as Kb from '../../common-adapters'
+import * as Styles from '../../styles'
+import * as TeamBuildingGen from '../../actions/team-building-gen'
+import {AllowedNamespace} from 'constants/types/team-building'
+import {requestIdleCallback} from '../../util/idle-callback'
+
+type EmailInputProps = {
+  namespace: AllowedNamespace
+}
+
+function checkValidEmail(str: string): boolean {
+  const emailRegex = /^(\S+@\S+\.\S+)$/
```

Comment:
Stole this regex from `libkb/checkers.go`. Couldn't find anything client side that did email validation, and I didn't want to send an RPC on each key to validate the email...